### PR TITLE
Filtering review fixes

### PR DESCRIPTION
**Pull request by jaleszek**
Bunch of fixed bugs found during the final testing the app/users lists; filtering.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -68,18 +68,27 @@ def after_initialize
...
-                    :email,
-                    :first_name,
-                    :last_name,
-                    :company_name
-                  ],
-                  using: {
-                    tsearch: {
-                      prefix: true
+  begin :searching
+    pg_search_scope :basic_search,
```
Comment (matthewford) [inline app/models/user.rb]: Inconsistent indentation detected.

File: app/models/user.rb [hunk 2]
```diff
@@ -68,18 +68,27 @@ def after_initialize
...
+                      :company_name
+                    ],
+                    using: {
+                      tsearch: {
+                        prefix: true
+                      }
                     }
-                  }
+    # TODO: take into consideration forcing NULL for all attributes.
+    nilify_blanks only: [
```
Comment (matthewford) [inline app/models/user.rb]: Inconsistent indentation detected.

File: app/search/form_answer_search.rb [hunk 3]
```diff
@@ -58,7 +58,7 @@ def filter_by_sub_status(scoped_results, value)
           "LEFT OUTER JOIN palace_attendees ON palace_attendees.palace_invite_id = palace_invites.id"
         ).where("palace_invites.id IS NULL OR palace_attendees.id IS NULL")
       when "missing_corp_responsibility"
-        out = out.where("document @> '\"corp_responsibility_form\"=>\"declare_now\"'::hstore")
+        out = out.where("form_answers.document @> '\"corp_responsibility_form\"=>\"declare_now\"'::hstore")
```
Comment (matthewford) [inline app/search/form_answer_search.rb]: Line is too long. [107/100]

**Comment by RuslanHamidullin** [comment]
Good to merge :+1:

**Pull request merged**